Source/Core/Core/ConfigManager.cpp

```diff
@@ -98,6 +99,42 @@ void SConfig::LoadSettings()
   Config::Load();
 }
 
+const std::string& SConfig::GetGameID() const
+{
+  std::lock_guard<std::mutex> lock(m_metadata_lock);
+  return m_game_id;
+}
```

Comment:
If you lock the read then you also need to lock the write, otherwise there's no point locking the read.

Reply:
Also you can't return a reference here, that defeats the point of the locking since the reference would then be accessed outside the scope of the lock. You need to return a copy.

@JosJuice Do you have any performance concerns about this? I'm not sure if the metadata here is ever accessed in anything timing-critical.

Reply:
This is not resolved.

Reply:
Oops, I didn't push my changes for this, sorry about that.

Reply:
I'm not aware of anything timing-critical that accesses this.